### DISCUSSION TITLE
Need help with NextJS + Electron

### DISCUSSION THREAD

**Discussion by alancunha26** [vercel/next.js#17543] [Help]
### The problem
Hello guys, I have an application that will be desktop supported (using native modules) and will be published in web as well. The main problem is that I would like to have all the befits of using next for developing the project, because web would be SEO friendly with all the revalidation stuff, but my app need to run in Electron too, because I need access to all desktop tools and apis that Electron provides.

I made a lot of research about this matter, and find some solutions, but nothing that looks really production friendly. Maybe I am being too picky and should leave next and use plain react, but I really want to my web application be server side rendered. Maybe someone here could help me with this or give me another solutions.

After some thinking and researching I believe that I could write the application with react-router and use nextJS only in the web module for SSR, while I render the app with react-router in the renderer layer of electron. Besides that I would be happy if someone maybe could help me with this specific use case and give me a less tricky solution.

Thanks for your help and sorry for my english!

### Features needed:
- Shared code between web and desktop;
- Server side render for web (SEO);
- Export native modules and features from electron to renderer;

### References
- https://github.com/saltyshiomix/nextron
- https://github.com/vercel/next.js/tree/master/examples/with-electron-typescript
- https://leo.im/2017/electron-next

**Comment by mrgoonie** [comment]
This one seems to be promising: https://github.com/saltyshiomix/nextron

**Comment by thoaihuynh2509** [comment]
@alancunha26 Have you found the solution yet?

**Comment by akash07k** [comment]
@alancunha26  have you came across any solution?
